pipeline-model-extensions/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/agent/CheckoutScript.groovy

```diff
@@ -44,17 +44,24 @@ public class CheckoutScript implements Serializable {
     
     private static Closure checkoutAndRun(CpsScript script, DeclarativeAgent agent, Closure body) {
         return {
+            def checkoutMap
             if (agent.isDoCheckout() && agent.hasScmContext(script)) {
                 if (!agent.inStage) {
                     script.stage(SyntheticStageNames.checkout()) {
-                        script.checkout script.scm
+                        checkoutMap = script.checkout script.scm
                     }
                 } else {
                     // No stage when we're in a nested stage already
-                    script.checkout script.scm
+                    checkoutMap = script.checkout script.scm
                 }
             }
-            body.call()
+            if (checkoutMap != null && checkoutMap instanceof Map) {
```

Comment:
Either

```groovy
if (checkoutMap != null) { // assumed to be a Map
```

or

```groovy
if (checkoutMap instanceof Map) { // hence, not null
```

Probably you also want to avoid an extra block step if we are in legacy mode:

```groovy
if (checkoutMap instanceof Map && !checkoutMap.isEmpty()) {
```

which could also be simplified to just

```groovy
if (checkoutMap) { // handles null and empty maps
```